src/mumble/UserModel.cpp

```diff
@@ -1959,21 +1959,41 @@ QString UserModel::createDisplayString(const ClientUser &user, bool isChannelLis
 		friendTag = QString::fromLatin1("(%2)").arg(user.qsFriendName);
 	}
 
+	// Create a nickname-tag
+	QString nickname = user.getLocalNickname();
+	QString nicknameTag;
+	if (!nickname.isEmpty() && user.qsName.compare(nickname, Qt::CaseInsensitive) != 0) {
+		if (g.s.bShowNicknamesOnly) {
+			nicknameTag = QString::fromLatin1("%2").arg(nickname);
+		}
+		else {
+			nicknameTag = QString::fromLatin1("[%2]").arg(nickname);
+		}
+	}
+
 	// Create a tag that indicates the volume adjustments
 	QString volumeTag;
 	if (std::abs(localVolumeDecibel) > 0 && g.s.bShowVolumeAdjustments) {
 		volumeTag = QString::asprintf("|%+d|", localVolumeDecibel);
 	}
 
-	QString displayString = user.qsName;
+	QString displayString;
+	if (!g.s.bShowNicknamesOnly || nicknameTag.isEmpty()) {
+		displayString = user.qsName;
+	}
 
 	if (!friendTag.isEmpty()) {
 		displayString += " " + friendTag;
 	}
 
+	if (!nicknameTag.isEmpty()) {
+		displayString += " " + nicknameTag;
+	}
```

Comment:
In case Mumble will only show the nickname and no friend tag, this will introduce a spurious space in the user's name.

And in case there is a friend tag, the friend tag will be shown before the user's nickname (which in case the actual name is not shown, can be considered the "normal" username from a user's perspective). That probably looks a bit odd.

I think I would handle the nickname-only setting entirely where the normal name is handled.

Something like this:

```
QString displayString;
if (!g.s.bSHowNicknameOnly || nickname.isEmpty()) {
    displayString += user.qsName;
} else {
    displayString += nickname;
}

// friend tag

if (!g.s.bShowNicknamesOnly && !nickname.isEmpty() && user.qsName.compare(nickname, Qt::CaseInsensitive) != 0) {
    displayString += " " + QString::fromLatin1("[%1]").arg(nickname);
}
```
Note how this completely eliminates the need for the `nicknameTag` helper variable :point_up: